manifests/plugin/discovery.pp

```diff
@@ -15,16 +20,21 @@
 # $image_name::   tarball with images
 #
 class foreman_proxy::plugin::discovery (
+  Boolean $enabled = true,
+  Optional[String] $version = undef,
+  Foreman_proxy::ListenOn $listen_on = 'https',
   Boolean $install_images = $foreman_proxy::plugin::discovery::params::install_images,
   Stdlib::Absolutepath $tftp_root = $foreman_proxy::plugin::discovery::params::tftp_root,
   Stdlib::HTTPUrl $source_url = $foreman_proxy::plugin::discovery::params::source_url,
   String $image_name = $foreman_proxy::plugin::discovery::params::image_name,
 ) inherits foreman_proxy::plugin::discovery::params {
-  foreman_proxy::plugin { 'discovery':
+  foreman_proxy::plugin::module { 'discovery':
+    enabled   => $enabled,
+    feature   => 'Discovery',
```

Comment:
Technically redundant, but doesn't hurt.

Reply:
I would be happy to remove that... I was copying the style of other classes under the same path.

Reply:
https://github.com/theforeman/puppet-foreman_proxy/pull/773